lib/internal/streams/from.js

```diff
@@ -53,18 +50,22 @@ function from(Readable, iterable, opts) {
   };
 
   readable._destroy = function(error, cb) {
-    if (needToClose) {
-      needToClose = false;
-      close().then(
-        () => process.nextTick(cb, error),
```

Comment:
Was there any reason I missed for this to run in `nextTick` here? 

(Note that `then` callbacks already defer a microtask)

Reply:
If the callback throws, it gets captured by the promise, which will reject and cause an unhandledRejection.

Reply:
@mcollina good point - is that still an issue now that unhandled rejections and uncaught exceptions behave the same way and close the process?

(Genuinely asking, since there is a small semantic difference here eventhough in both cases we wait for a microtick before emitting the error - the `unhandledRejection` event is emitted before `uncaughtException` without the nextTick)

Reply:
I would keep the nextTick if you plan to have this backported... which is something I think we should do!

Reply:
Ok, I will revert and add a comment.

Reply:
Done